source/common/router/router.cc

```diff
@@ -2386,13 +2386,13 @@ void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or
         *cluster_->lrsReportMetricNames(), *orca_load_report, upstream_host->loadMetricStats());
   }
   if (host_lb_policy_data.has_value()) {
-    ENVOY_LOG(trace, "orca_load_report for {} report = {}", upstream_host->address()->asString(),
-              (*orca_load_report).DebugString());
+    ENVOY_STREAM_LOG(trace, "orca_load_report for {} report = {}", *callbacks_,
+                     upstream_host->address()->asString(), (*orca_load_report).DebugString());
     const absl::Status status =
         host_lb_policy_data->onOrcaLoadReport(*orca_load_report, callbacks_->streamInfo());
     if (!status.ok()) {
-      ENVOY_STREAM_LOG(error, "Failed to invoke OrcaLoadReportCallbacks: {}", *callbacks_,
-                       status.message());
+      ENVOY_LOG_EVERY_POW_2(error, "Failed to invoke OrcaLoadReportCallbacks: {}",
```

Comment:
If this log-line appears many times, could it be that this is not an error, but WAI?
If this is a valid state to be in, then it should be explicitly stated in the code, and no error should be emitted, no?

Reply:
@adisuissa - do you suggest to just make it a `trace` and never log as an `error`? I was thinking about using this for debugging purposes. Maybe also log the upstream address and load report debug string?

@ggreenway - That is an interesting suggestion! 
- Is adding a metric expensive? 
- Would this metric be added to `ALL_ROUTER_STATS` or elsewhere? 
- Are those stats grouped by cluster? 
- Would it make sense to also add a metric for successful path as well?

Reply:
My question is whether getting to this `if (!status.ok()) {` is a valid state?
If it is, then it shouldn't be a warning/error. If it isn't, then why would this log line be omitted so many times that the logs should be power of 2?

Reply:
Does the log message contain useful information about the cause/reason, or is it basically just saying that it happened? If the latter, I think a metric is more appropriate. If the former, I wonder if we want to make it time-based instead of pow_2, because if the reason is not always the same, after a certain number of logs it will log so infrequently that you'll never see the reason again.